lib/charms/kubernetes_charm_libraries/v0/volumes.py

```diff
@@ -0,0 +1,415 @@
+    """Class to be instantiated by charms requiring requested volumes."""
+
+    def __init__(
+        self,
+        namespace: str,
+        application_name: str,
+        unit_name: str,
+        container_name: str,
+    ):
+        """Constructor for the KubernetesVolumesPatchLib.
+
+        Args:
+            namespace: Namespace name
+            application_name: Charm application name
+            unit_name: Unit name
+            container_name: Container name
+        """
+        self.kubernetes = KubernetesClient(namespace=namespace)
+        self.model_name = namespace
+        self.application_name = application_name
+        self.unit_name = unit_name
+        self.container_name = container_name
+
+    def add_requested_volumes(self, requested_volumes, container_name) -> None:
+        """Creates volumes and patches statefulset.
+
+        Args:
+            requested_volumes: Iterable of volumes to add to the statefulset
+            container_name: Container name
+        """
+        self.kubernetes.patch_volumes(
+            statefulset_name=self.application_name,
+            requested_volumes=requested_volumes,
+            container_name=container_name,
+        )
+
+    def remove_requested_volumes(
+        self,
+        requested_volumes: Iterable[RequestedVolume],
+        container_name: str,
+    ) -> None:
+        """Deletes volumes from statefulset and pod.
+
+        Args:
+            requested_volumes: Iterable of volumes to add to the statefulset
+            container_name: Container name
+        """
+        self.kubernetes.remove_volumes(
+            statefulset_name=self.application_name,
+            requested_volumes=requested_volumes,
+            container_name=container_name,
+        )
+
+    def _pod_is_ready(self, requested_volumes: Iterable[RequestedVolume]) -> bool:
+        """Returns whether pod is ready with requested volumes.
+
+        Args:
+            requested_volumes: Iterable of volumes to add to the statefulset
+        """
+        return self.kubernetes.pod_is_ready(
+            pod_name=self._pod,
+            requested_volumes=requested_volumes,
+            container_name=self.container_name,
+        )
+
+    @property
+    def _pod(self) -> str:
+        """Name of the unit's pod.
+
+        Returns:
+            str: A string containing the name of the current unit's pod.
+        """
+        return "-".join(self.unit_name.rsplit("/", 1))
+
+    def _statefulset_is_patched(self, requested_volumes: Iterable[RequestedVolume]) -> bool:
+        """Returns whether statefuset is patched with requested volumes.
+
+        Args:
+            requested_volumes: Iterable of volumes to add to the statefulset
+        """
+        return self.kubernetes.statefulset_is_patched(
+            statefulset_name=self.application_name,
+            requested_volumes=requested_volumes,
+        )
+
+    def is_ready(self, requested_volumes: Iterable[RequestedVolume]) -> bool:
```

Comment:
I think 'is_ready' can be confusing here as well. I wouldn't know what action it refers to: addition or removal of the volumes? 
Maybe something like is_addition_of_volumes_finished?